crates/burn-import/src/onnx/op_configuration.rs

```diff
@@ -748,43 +748,66 @@ pub fn layer_norm_config(node: &Node) -> (LayerNormConfig, bool) {
 /// Create a PadConfig from the attributes of the node
 pub fn pad_config(node: &Node) -> PadConfig {
     fn get_pads(node: &Node) -> Vec<usize> {
+        if node.inputs.len() != 3 {
+            panic!("Pad: must provide three inputs")
+        }
+
         let input_dim = match &node.inputs.first().unwrap().ty {
             ArgType::Tensor(tensor) => tensor.dim,
-            _ => panic!("Only tensor input is valid"),
+            _ => panic!("Pad: Only tensor input is valid"),
         };
+
         let pads: Vec<usize> = match &node.inputs[1].value {
             Some(Data::Int64s(shape)) => shape
                 .iter()
                 .map(|&x| {
                     if x < 0 {
                         // TODO: support negative pads
-                        panic!("Negative pad is not supported");
+                        panic!("Pad: Negative pad is not supported");
                     }
                     x as usize
                 })
                 .collect(),
-            _ => panic!("Pads data type must be int64"),
+            _ => panic!("Pad: pads data type must be int64"),
         };
 
-        assert_eq!(pads.len(), input_dim * 2);
+        if pads.len() != input_dim * 2 {
+            panic!("Pad: pads should be a 1D tensor of shape [2 * num_axes]");
+        }
+        // TODO: Burn's pad should support 1D tensor
+        if input_dim < 2 {
+            panic!("Pad: input tensor should be rank 2 or higher");
+        }
+
+        let left_index = input_dim - 1;
+        let top_index = input_dim - 2;
+        let right_index = pads.len() - 1;
+        let bottom_index = pads.len() - 2;
+        let index_list = [left_index, top_index, right_index, bottom_index];
 
```

Comment:
This seems to accept pads with pads.len() > 4 but Burn API only accepts 4. Right?

Reply:
If pads.len() > 4, this code checks if the user wants to pad any dimension than the last two:

yes -> panic
no -> the pad operation is doable, and it calculates l, t, r, b

Reply:
The way the padding is specified for the ONNX Pad op is a bit different since it supports padding more than just the last two dimensions. So we have to parse the provided `pads` to make sure it is valid, and then return the Burn valid padding with only 4 values for (left, right, top, bottom).

I think the check is valid. For a 4d input where we should only accept padding the last two dimensions, it means that `pads` should be `[0, 0, x2_begin, x3_begin, 0, 0, x2_end, x3_end]`. For the last two dimensions expected to be `[H, W]`, this means that `x2_begin = top`, `x3_begin = left`, `x2_end = bottom` and `x3_end = right`.